esphome/components/max6921/display.cpp

```diff
@@ -0,0 +1,953 @@
+  memset(this->out_buf_, 0, this->out_buf_size_);
+
+  // find smallest segment DOUT number...
+  this->seg_out_smallest_ = 19;
+  for (unsigned char i : this->seg_to_out_map_) {
+    if (i < this->seg_out_smallest_)
+      this->seg_out_smallest_ = i;
+  }
+  ESP_LOGCONFIG(TAG, "Display smallest DOUT number: %u", this->seg_out_smallest_);
+
+  // calculate refresh period for 60Hz
+  this->refresh_period_us_ = 1000000 / 60 / this->num_digits_;
+  ESP_LOGCONFIG(TAG, "Set display refresh period: %" PRIu32 "us for %u digits @ 60Hz", this->refresh_period_us_,
+                this->num_digits_);
+
+  /* Setup display refresh.
+   * Using a timer is not an option, because the WiFi component uses timer as
+   * well, which leads to unstable refresh cycles (flickering). Therefore a
+   * thread on 2nd MCU core is used.
+   */
+  xTaskCreatePinnedToCore(&Max6921Display::display_refresh_task,
+                          "display_refresh_task",  // name
+                          2048,                    // stack size
+                          this,                    // pass component pointer as task parameter pv
+                          1,                       // priority (one above IDLE task)
+                          nullptr,                 // handle
+                          1                        // core
+  );
+
+  // ESP_LOGCONFIG(TAG, "Display mode: %u", this->mode);
+  // ESP_LOGCONFIG(TAG, "Display text effect: %u", this->disp_text_.effect);
+}
+
+void HOT Max6921Display::display_refresh_task(void *pv) {
+  Max6921Display *display = (Max6921Display *) pv;
+  static uint count = display->num_digits_;
+  static uint current_pos = 1;
+
+  if (display->refresh_period_us_ == 0) {
+    ESP_LOGE(TAG, "Invalid display refresh period -> using default 2ms");
+    display->refresh_period_us_ = 2000;
+  }
+
+  while (true) {
+    // one-time verbose output for all positions after any content change...
+    if (display->disp_text_.content_changed) {
+      count = 0;
+      display->disp_text_.content_changed = false;
+    }
+    if (count < display->num_digits_) {
+      count++;
+      ESP_LOGVV(TAG, "%s(): SPI transfer for position %u: 0x%02x%02x%02x", __func__, current_pos,
+                display->out_buf_[(current_pos - 1) * 3], display->out_buf_[(current_pos - 1) * 3 + 1],
+                display->out_buf_[(current_pos - 1) * 3 + 2]);
+    }
+
+    // write MAX9621 data of current display position...
+    display->max6921_->write_data(&display->out_buf_[(current_pos - 1) * 3], 3);
+
+    // next display position...
+    if (++current_pos > display->num_digits_)
+      current_pos = 1;
+
+    delayMicroseconds(display->refresh_period_us_);
+  }
+}
+
+void Max6921Display::dump_config() {
+  char seg_name[3];
+
+  // display segment to DOUTx mapping...
+  for (uint i = 0; i < this->seg_to_out_map_.size(); i++) {
+    if (i < 7) {
+      seg_name[0] = 'a' + i;
+      seg_name[1] = 0;
+    } else
+      strncpy(seg_name, "dp", sizeof(seg_name));
```

Comment:
All branches should have braces
```suggestion
    } else {
      strncpy(seg_name, "dp", sizeof(seg_name));
    }
```